src/azure-cli/azure/cli/command_modules/appconfig/_snapshot_custom_client.py

```diff
@@ -0,0 +1,513 @@
+    if_match=None,
+    if_none_match=None,
+    sync_token=None,
+    **kwargs
+):
+
+    request_body = {}
+
+    if not filters or len(filters) < 1:
+        raise ValueError("There should be at least one filter specified.")
+
+    request_body["filters"] = filters
+
+    if composition_type is not None:
+        if composition_type not in ("all", "group_by_key"):
+            raise ValueError("Value should either be 'all' or 'group_by_key'.")
+
+        request_body["composition_type"] = composition_type
+
+    if retention_period is not None:
+        if not isinstance(retention_period, int) or retention_period < 0:
+            raise ValueError("Retention period value should be a non-negative integer value.")
+
+        request_body["retention_period"] = retention_period
+
+    if tags:
+        request_body["tags"] = tags
+
+    return _build_request(
+        "/snapshots/{name}",
+        RequestMethod.PUT,
+        path_arguments={"name": name},
+        json=request_body,
+        if_match=if_match,
+        if_none_match=if_none_match,
+        sync_token=sync_token,
+        **kwargs
+    )
+
+
+def _format_url_section(template, **kwargs):
+    components = template.split("/")
+    while components:
+        try:
+            return template.format(**kwargs)
+        except KeyError as key:
+            formatted_components = template.split("/")
+            components = [c for c in formatted_components if "{}".format(key.args[0]) not in c]
+            template = "/".join(components)
+
+
+class AppConfigSnapshotClient:
+
+    def __init__(self, appConfigClient):
+        appConfiguration = appConfigClient._impl
+        self._serializer = appConfiguration._serialize
+        self._deserializer = appConfiguration._deserialize
+        self._sync_token = appConfiguration._config.sync_token
+        self._endpoint = appConfiguration._config.endpoint
+        self._client = appConfiguration._client
+
+    @classmethod
+    def from_connection_string(cls, connection_string):
+        return cls(AzureAppConfigurationClient.from_connection_string(connection_string))
+
+    def begin_create_snapshot(
+            self,
+            name,
+            filters,
+            composition_type=None,
+            retention_period=None,
+            tags=None,
+            if_match=None,
+            if_none_match=None,
+            **kwargs):
+        """
+        Poll after a given interval (default 5s) to ensure that the snapshot is in 'ready' status.
+        The request times out after 30s by default unless specified otherwise.
+        """
+        timeout = kwargs.pop("timeout", 30)
+        polling_interval = kwargs.pop("polling_interval", 5)
+
+        from datetime import datetime
+        from knack.log import get_logger
+        
+        def _get_elapsed_time(start_time: datetime):
+            return (datetime.now() - start_time).total_seconds()
+
+        logger = get_logger(__name__)
+        
+        logger.warning("Starting...")
```

Comment:
This was tried however, the logged data was not shown in the cli as the default verbosity of the cli logs above the level of "information".  I used `warning` according to the guidelines [here ](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#coding-practices:~:text=Log%20to%20logger.error()%20or%20logger.warning()%20for%20user%20messages%3B%20do%20not%20use%20the%20print()%20function) since directly using the `print` function is discouraged.